src/offline/client.cc

```diff
@@ -456,10 +465,15 @@ PostRunAction run(const Config& cfg_in, std::shared_ptr<HttpInterface> docker_cl
   if (!rollback_target.IsValid()) {
     // If either a device failed to boot on a new device or ostree hasn't changed but new version Apps failed to start,
     // then the current version is actually "rollback" Target we need to switch to, which is effectivelly syncing Apps
-    rollback_target = current_target;
+    if (current_target.IsValid() && current_target.filename() != "unknown") {
+      rollback_target = current_target;
+    } else {
+      // something really bad has happened
```

Comment:
will the logs already include some context - or do we need to say something additional here?